src/OnlineRetailPortal.Mock/MockProductStore.cs

```diff
@@ -69,8 +68,18 @@ public async Task<GetProductsStoreResponse> GetProductsAsync(GetProductsStoreReq
 
             response = await Task.Run(() =>
             {
-                List<Product> products = productList.Where(y => y.Status == Status.Active).OrderBy(x => x.PostDateTime).Take(50).ToList();
-                response = products.ToGetProductsStoreResponse();
+                List<Product> products = productList.OrderByDescending(x => x.PostDateTime).ToList();
+                int startIndex = (request.PagingInfo.PageNumber - 1) * (request.PagingInfo.PageSize);
+                int endIndex = startIndex + request.PagingInfo.PageSize - 1;
+                List<Product> newProductList = new List<Product>();
+                for (int i = startIndex; i <= endIndex; i++)
+                {
+                    if (i == products.Count())
+                        break;
+                    newProductList.Add(products[i]);
+                }
+                request.PagingInfo.TotalPages = productList.Count() / request.PagingInfo.PageSize+ productList.Count()% request.PagingInfo.PageSize;
+                response = newProductList.ToGetProductsStoreResponse(request.PagingInfo.PageNumber,request.PagingInfo.PageSize, request.PagingInfo.TotalPages);
```

Comment:
Why are we not sending request.PagingInfo and sending every field inside it seperately instead.

Reply:
done